### DISCUSSION TITLE
[Feature] Security Support for Non-VPN Deployment + Notification When This is The Case

### DISCUSSION THREAD

**Discussion by rothn** [immich-app/immich#13008] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### Request
I would be very grateful if you all could provide a timeframe for a release that can safely be exposed to the Internet when properly configured and a mechanism (e.g. mailing list, Matrix) to be alerted when this is the case.

### Background
Canonically, I've seen people set things up something like "[Internet]--[CDN]--[NetworkFrontend]--[Service]". The CDN handles delivery with their own backbone and PoPs and may also help with DDOS protection. The network frontend handles encryption and routing to the service. And the service (Immich) does something useful.

Because this is simple and known to me, I prefer to set up my own stuff this way too. When I heard about Immich, I was excited to try it out, but concerned about security. Indeed, upon checking the documentation, I found the phrase "Immich is under very active development and the existence of severe security vulnerabilities cannot be ruled out." To me, this means making it easily available to myself in the way described above (and to an extent in your documentation under Remote Access, Option 3") is probably not what I want to do right now. At the same time, this is by far my preferred alternative.

I fiddled with Wireguard and routing rules for a day and was not able to get that working. This is probably just due to my lack of familiarity, but despite the protocol's simplicity, getting it working in practice seems really complicated. I don't want to spend any more time figuring this out for just one service. I'm probably not the only one.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
When Immich is "stable" we will release v2.0.0 and remove the active development/expect bugs warning. Of course we do our best to prevent any security issues, but ultimately exposing Immich safely is in your hands. If you know what you're doing then you can probably do it no problem right now, if you don't know what you're doing then you probably shouldn't even after stable.

**Reply by rothn** [reply]
I guess I'll monitor it. Any stuff I should turn off that will still let the web UI and app work? Would it be better to turn the web UI off for external requests with some rules, or does it just use the API internally? And if so, which paths should I block?

**Reply by rothn** [reply]
In case anyone else comes across my post, I ended up using https://github.com/coreruleset/modsecurity-crs-docker while waiting for more information here
